test/spanner.ts

```diff
@@ -1486,6 +1489,69 @@ describe('Spanner with mock server', () => {
         });
       });
     });
+
+    describe('LeaderAwareRouting', () => {
+      let spannerWithLARDisabled: Spanner;
+      let instanceWithEnvVar: Instance;
+
+      function newTestDatabaseWithLARDisabled(
+        options?: SessionPoolOptions,
+        queryOptions?: IQueryOptions
+      ): Database {
+        return instanceWithEnvVar.database(
+          `database-${dbCounter++}`,
+          options,
+          queryOptions
+        );
+      }
+
+      before(() => {
+        spannerWithLARDisabled = new Spanner({
+          servicePath: 'localhost',
+          port,
+          sslCreds: grpc.credentials.createInsecure(),
+          routeToLeaderEnabled: false,
+        });
+        // Gets a reference to a Cloud Spanner instance and database
+        instanceWithEnvVar = spannerWithLARDisabled.instance('instance');
+      });
+
+      it('should execute with leader aware routing enabled in a read/write transaction', async () => {
+        const database = newTestDatabase();
+        await database.runTransactionAsync(async tx => {
+          await tx!.runUpdate({
+            sql: insertSql,
+          });
+          return await tx.commit();
+        });
+        await database.close();
+        spannerMock.getMetadata().forEach(metadata => {
+          if (metadata.get(LEADER_AWARE_ROUTING_HEADER)[0] !== undefined) {
+            assert.strictEqual(
+              metadata.get(LEADER_AWARE_ROUTING_HEADER)[0],
```

Comment:
Could we also add an assertion that we have seen at least one metadata instance with leader-aware-routing? (The current assertion also succeeds if `metadata.get(LEADER_AWARE_ROUTING_HEADER)[0] === undefined` for all requests)